curvefs/src/mds/fs_manager.cpp

```diff
@@ -790,6 +790,39 @@ FSStatusCode FsManager::GetFsInfo(const std::string& fsName, uint32_t fsId,
     return FSStatusCode::OK;
 }
 
+FSStatusCode FsManager::UpdateFsInfo(
+    const ::curvefs::mds::UpdateFsInfoRequest* req) {
+    const auto& fsName = req->fsname();
+    NameLockGuard lock(nameLock_, fsName);
+
+    // 1. query fs
+    FsInfoWrapper wrapper;
+    FSStatusCode ret = fsStorage_->Get(fsName, &wrapper);
+    if (ret != FSStatusCode::OK) {
+        LOG(WARNING) << "UpdateFsInfo fail, get fs fail, fsName = " << fsName
+                     << ", errCode = " << FSStatusCode_Name(ret);
+        return ret;
+    }
+
+    // 2. check fs status
+    FsStatus status = wrapper.GetStatus();
+    if (status == FsStatus::NEW) {
+        LOG(WARNING) << "UpdateFsInfo fs is not inited, fsName = " << fsName;
+        return FSStatusCode::NOT_INITED;
+    } else if (status == FsStatus::DELETING) {
+        LOG(WARNING) << "UpdateFsInfo fs is in deleting, fsName = " << fsName;
+        return FSStatusCode::UNDER_DELETING;
+    }
+
+    // 3. update fs info
+    // set capacity
+    if (req->has_capacity()) {
```

Comment:
check this at beginning, the step 1 and 2 are not required if capacity not exist.

Reply:
ok i will add parameter check in MdsServiceImpl::UpdateFsInfo

Reply:
It has been checked in [mds servers](https://github.com/h0hmj/curve/blob/d5fe7de44b0b6743210ba5579c1b981567cd3aae/curvefs/src/mds/mds_service.cpp#L240). Do I still need to check here?